File: Samples/MarbleMaze_4_0/Source/EX2_Polishing/End/MarbleMazeGame/MarbleMazeGame/Screens/BackgroundScreen.cs
#region File Description
//-----------------------------------------------------------------------------
// BackgroundScreen.cs
//
// Microsoft XNA Community Game Platform
// Copyright (C) Microsoft Corporation. All rights reserved.
//-----------------------------------------------------------------------------
#endregion

#region Using Statements
using System;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using GameStateManagement; 
#endregion

namespace MarbleMazeGame
{
    class BackgroundScreen : GameScreen
    {
        #region Fields
        Texture2D background; 
        #endregion

        #region Initialization
        public BackgroundScreen()
        {
            TransitionOnTime = TimeSpan.FromSeconds(0.0);
            TransitionOffTime = TimeSpan.FromSeconds(0.5);
        } 
        #endregion

        #region Loading
        /// <summary>
        /// Load screen resources
        /// </summary>
        public override void LoadContent()
        {
            background = Load<Texture2D>(@"Images\titleScreen");
        } 
        #endregion

        #region Update
        /// <summary>
        /// Update the screen
        /// </summary>
        /// <param name="gameTime">The game time</param>
        /// <param name="otherScreenHasFocus"></param>
        /// <param name="coveredByOtherScreen"></param>
        public override void Update(GameTime gameTime, bool otherScreenHasFocus,
                            bool coveredByOtherScreen)
        {
            base.Update(gameTime, otherScreenHasFocus, false);
        } 
        #endregion

        #region Render
        /// <summary>
        /// Renders the screen
        /// </summary>
        /// <param name="gameTime"></param>
        public override void Draw(GameTime gameTime)
        {
            SpriteBatch spriteBatch = ScreenManager.SpriteBatch;            

            spriteBatch.Begin();

            // Draw background
            spriteBatch.Draw(background, new Vector2(0, 0),
                 Color.White * TransitionAlpha);

            spriteBatch.End();
        } 
        #endregion
    }
}

File: Samples/CardsStarterKit_4_0_VB/CardsStarterKit_4_0/CardsGame/CardsGame/Screens/BackgroundScreen.vb
#Region "File Description"
'-----------------------------------------------------------------------------
' BackgroundScreen.vb
'
' Microsoft XNA Community Game Platform
' Copyright (C) Microsoft Corporation. All rights reserved.
'-----------------------------------------------------------------------------
#End Region

Imports System.Text
Imports Blackjack.GameStateManagement


Friend Class BackgroundScreen
    Inherits GameScreen
    Private background As Texture2D
    Private safeArea As Rectangle

    ''' <summary>
    ''' Initializes a new instance of the screen.
    ''' </summary>
    Public Sub New()
        TransitionOnTime = TimeSpan.FromSeconds(0.0)
        TransitionOffTime = TimeSpan.FromSeconds(0.5)
    End Sub

#Region "Loading"
    ''' <summary>
    ''' Load graphics content for the screen.
    ''' </summary>
    Public Overrides Sub LoadContent()
        background = ScreenManager.Game.Content.Load(Of Texture2D)("Images\titlescreen")
        safeArea = ScreenManager.Game.GraphicsDevice.Viewport.TitleSafeArea
        MyBase.LoadContent()
    End Sub
#End Region

#Region "Update and Render"
    ''' <summary>
    ''' Allows the screen to run logic, such as updating the transition position.
    ''' Unlike HandleInput, this method is called regardless of whether the screen
    ''' is active, hidden, or in the middle of a transition.
    ''' </summary>
    ''' <param name="gameTime"></param>
    ''' <param name="otherScreenHasFocus"></param>
    ''' <param name="coveredByOtherScreen"></param>
    Public Overrides Sub Update(ByVal gameTime As GameTime, ByVal otherScreenHasFocus As Boolean, ByVal coveredByOtherScreen As Boolean)
        MyBase.Update(gameTime, otherScreenHasFocus, False)
    End Sub

    ''' <summary>
    ''' This is called when the screen should draw itself.
    ''' </summary>
    ''' <param name="gameTime"></param>
    Public Overrides Sub Draw(ByVal gameTime As Microsoft.Xna.Framework.GameTime)
        ScreenManager.SpriteBatch.Begin()

        ScreenManager.SpriteBatch.Draw(background, ScreenManager.GraphicsDevice.Viewport.Bounds, Color.White * TransitionAlpha)

        ScreenManager.SpriteBatch.End()

        MyBase.Draw(gameTime)
    End Sub
#End Region
End Class
